Dockerfile

```
@@ -19,28 +23,96 @@ COPY ./app/img/ /source/app/img/
 
 # Create the base image
-FROM debian:12.2-slim@sha256:93ff361288a7c365614a5791efa3633ce4224542afb6b53a1790330a8e52fc7d
+FROM debian:12.2-slim@sha256:93ff361288a7c365614a5791efa3633ce4224542afb6b53a1790330a8e52fc7d as base
+
+# Add curl to the base image (7.88.1-10+deb12u5)
+# Add jq to the base image (1.6-2.1)
+RUN apt-get update && apt-get install -y curl=7.88.1-10+deb12u5 jq=1.6-2.1
 
 # Install kubo and initialize ipfs
 COPY --from=ipfs-kubo /usr/local/bin/ipfs /usr/local/bin/ipfs
 
+# Copy app's build output and initialize IPFS api
+COPY --from=builder /source/app /export
 RUN ipfs init
+RUN ipfs add --cid-version 1 --quieter --only-hash -r /export > ipfs_hash.txt
 
-# Copy lunaria build output
-COPY --from=builder /source/app /export
+# --------------------------------------------------------
+# Publish Script: Option to host app locally or on nft.storage
+# --------------------------------------------------------
+
+COPY <<'EOF' /entrypoint.sh
+#!/bin/sh
+set -e
+
+if [ $# -ne  1 ]; then
+  echo "Example usage: docker run --rm ghcr.io/darkflorist/lunaria:latest [docker-host|nft.storage]"
+  exit  1
+fi
+
+case $1 in
+
+  docker-host)
+    # Show the IFPS build hash
+    echo "Build Hash: $(cat /ipfs_hash.txt)"
 
-# add the build output to IPFS and write the hash to a file
-RUN ipfs add --cid-version 1 --quieter --only-hash --recursive /export > ipfs_hash.txt
+    # Determine the IPV4 address of the docker-hosted IPFS instance
+    IPFS_IP4_ADDRESS=$(getent ahostsv4 host.docker.internal | grep STREAM | head -n 1 | cut -d ' ' -f 1)
 
-# print the hash for good measure in case someone is looking at the build logs
-RUN cat ipfs_hash.txt
+    echo "Adding files to docker running IPFS at $IPFS_IP4_ADDRESS"
+		IPFS_HASH=$(ipfs add --api /ip4/$IPFS_IP4_ADDRESS/tcp/5001 --cid-version 1 --quieter -r /export)
+    echo "Uploaded Hash: $IPFS_HASH"
+    ;;
 
-# this entrypoint file will execute `ipfs add` of the build output to the docker host's IPFS API endpoint, so we can easily extract the IPFS build out of the docker image
-RUN printf '#!/bin/sh\nipfs --api /ip4/`getent ahostsv4 host.docker.internal | grep STREAM | head -n 1 | cut -d \  -f 1`/tcp/5001 add --cid-version 1 -r /export' >> entrypoint.sh
-RUN chmod u+x entrypoint.sh
+  nft.storage)
+    if [ -z $NFTSTORAGE_API_KEY ] || [ $NFTSTORAGE_API_KEY = "" ]; then
+      echo "NFTSTORAGE_API_KEY environment variable is not set";
+      exit  1;
+    fi
+
+    # Show the IFPS build hash
+    echo "Build Hash: $(cat /ipfs_hash.txt)"
+
+    # Create a CAR archive from build hash
+    echo "Uploading files to nft.storage..."
+		IPFS_HASH=$(ipfs add --cid-version 1 --quieter -r /export)
+    ipfs dag export $IPFS_HASH > output.car
+
+    # Upload the entire directory to nft.storage
+		UPLOAD_RESPONSE=$(curl \
+			-X POST -H "Authorization: Bearer $NFTSTORAGE_API_KEY" \
+			-H "Content-Type: application/car" \
+			--data-binary @output.car \
+			-s \
+			https://api.nft.storage/upload)
```

Recommend using verbose parameters for enhanced clarity.
```suggestion
		UPLOAD_RESPONSE=$(curl \
			---request POST \
			--header "Authorization: Bearer $NFTSTORAGE_API_KEY" \
			--header "Content-Type: application/car" \
			--data-binary @output.car \
			--silent \
			https://api.nft.storage/upload)
```